packages/compose/src/hooks/use-viewport-match/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import useMediaQuery from '../use-media-query';
+
+/**
+ * Hash of breakpoint names with pixel width at which it becomes effective.
+ *
+ * @see _breakpoints.scss
+ *
+ * @type {Object}
```

Comment:
Nit:

```suggestion
 * @type {Object<string,number>}
```

Could even go further to define a custom type for the supported breakpoint keywords. This would mostly be helpful if it could be exposed to developers using the hook, to know what options they have available. But with our custom string syntax, I don't know that we can really add types details for that.

Example:

```diff
diff --git a/packages/compose/src/hooks/use-viewport-match/index.js b/packages/compose/src/hooks/use-viewport-match/index.js
index c20ecbde2..6193f6d6d 100644
--- a/packages/compose/src/hooks/use-viewport-match/index.js
+++ b/packages/compose/src/hooks/use-viewport-match/index.js
@@ -3,12 +3,16 @@
  */
 import useMediaQuery from '../use-media-query';
 
+/**
+ * @typedef {"huge"|"wide"|"large"|"medium"|"small"|"mobile"} WPBreakpoint
+ */
+
 /**
  * Hash of breakpoint names with pixel width at which it becomes effective.
  *
  * @see _breakpoints.scss
  *
- * @type {Object}
+ * @type {Object<WPBreakpoint,number>}
  */
 const BREAKPOINTS = {
 	huge: 1440,
@@ -19,6 +23,10 @@ const BREAKPOINTS = {
 	mobile: 480,
 };
 
+/**
+ * @typedef {">="|"<"} WPViewportOperator
+ */
+
 /**
  * Object mapping media query operators to the condition to be used.
  *
@@ -32,10 +40,8 @@ const CONDITIONS = {
 /**
  * Returns true if the viewport matches the given query, or false otherwise.
  *
- * @param {string} query Query string. Includes operator and breakpoint name,
- *                       space separated. Operator defaults to >=. The supported
- *                       breakpoint names are: mobile, small, medium, large, wide
- *                       and huge.
+ * @param {WPBreakpoint}       breakpoint      Breakpoint size name.
+ * @param {WPViewportOperator} [operator=">="] Viewport operator.
  *
  * @example
  *
@@ -46,11 +52,7 @@ const CONDITIONS = {
  *
  * @return {boolean} Whether viewport matches query.
  */
-const useViewportMatch = ( query ) => {
-	if ( query.indexOf( ' ' ) === -1 ) {
-		query = '>= ' + query;
-	}
-	const [ operator, breakpoint ] = query.split( ' ' );
+const useViewportMatch = ( breakpoint, operator = '>=' ) => {
 	const mediaQuery = `(${ CONDITIONS[ operator ] }: ${ BREAKPOINTS[ breakpoint ] }px)`;
 	return useMediaQuery( mediaQuery );
 };
```

Then we get some nice auto-completion and types checking:

![image](https://user-images.githubusercontent.com/1779930/70156148-7e132e80-1681-11ea-93fc-dd79782d9193.png)

Also, don't mind my recent types obsession 😄

Reply:
The more I think about this, the more I like the idea of changing to the `useViewportMatch( breakpoint, operator )` signature.

Reply:
Agreed, this is nicer, I updated.